server/sonar-server/src/main/java/org/sonar/server/ws/WebServiceEngine.java

```diff
@@ -95,7 +95,8 @@ public LocalResponse call(LocalRequest request) {
   public void execute(Request request, Response response) {
     try {
       ActionExtractor actionExtractor = new ActionExtractor(request.getPath());
-      WebService.Action action = getAction(actionExtractor.getController(), actionExtractor.getAction());
+      WebService.Action action = getAction(actionExtractor);
+      checkFound(action, "Unknown url : %s", request.getPath());
       if (request instanceof ValidatingRequest) {
```

Comment:
remark (not linked to your work): the Request class hierarchy would deserve to be simplified, it's very hard to understand what's called and where